dockerfiles/theia/src/add-extensions.js

```diff
@@ -98,38 +104,28 @@ function checkoutRepo(path, checkoutTarget) {
     }
 }
 
-function buildExtension(path) {
+function buildExtension(path,rootPath) {
     try {
+        console.log('Generate versions for extension: ', path);
+        spawnSync(`${DEFAULT_THEIA_ROOT}/versions.sh`, [], {cwd: `${path}`});
+        if (path !== rootPath) {
+            console.log("Removing parent files");
+            // cleanup files in root path
+            fs.unlinkSync(rootPath + '/package.json');
+            fs.unlinkSync(rootPath + '/lerna.json');
+        }
+        //spawnSync(`node`, [`${DEFAULT_THEIA_ROOT}/resolutions-provider.js`, `${rootPath}/package.json`], {cwd: `${rootPath}`});
         console.log('Building extension: ', path);
         const nodeModulesPath = `${DEFAULT_THEIA_ROOT}/node_modules`;
         // build extension, but use Theia node_modules to reuse dependencies and prevent growing docker image.
-        spawnSync(`yarn`, ['--modules-folder', nodeModulesPath, '--global-folder', nodeModulesPath, '--cache-folder', nodeModulesPath], {cwd: `${path}`, stdio:[0,1,2]});
+        spawnSync(`yarn`, ['--modules-folder', nodeModulesPath, '--global-folder', nodeModulesPath], {cwd: `${path}`, stdio:[0,1,2]});
     } catch (error) {
+        console.log(error);
```

Comment:
Maybe it would be better to use console.error here?

Reply:
sure

Reply:
fixed